common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -2094,6 +2084,16 @@ object CelebornConf extends Logging {
         "Allowed pattern is: `<host1>:<port1>[,<host2>:<port2>]*`")
       .createWithDefaultString(s"<localhost>:9097")
 
+  val MASTER_ENDPOINTS_RESOLVER: ConfigEntry[String] =
+    buildConf("celeborn.master.endpoints.resolver")
+      .categories("client", "worker")
+      .doc("Resolver class that can be used for dynamically discovering and updating the master endpoints. " +
```

Comment:
How could the user provide the custom reslover implementation? Especially for client?

Reply:
@SteNicholas How about below description, let me know if these looks ok.

```
  val MASTER_ENDPOINTS_RESOLVER: ConfigEntry[String] =
    buildConf("celeborn.master.endpoints.resolver")
      .categories("client", "worker")
      .doc("Resolver class that can be used for discovering and updating the master endpoints. This allows " +
        "users to provide a custom master endpoint resolver implementation. This is useful in environments " +
        "where the master nodes might change due to scaling operations or infrastructure updates. Clients " +
        "need to ensure that provided resolver class should be present in the classpath.")
      .version("0.6.0")
      .stringConf
      .createWithDefault("org.apache.celeborn.common.client.StaticMasterEndpointResolver")

  val MASTER_ENDPOINTS: ConfigEntry[Seq[String]] =
    buildConf("celeborn.master.endpoints")
      .categories("client", "worker")
      .doc("Endpoints of master nodes for celeborn clients to connect. Client uses resolver provided by" +
        s"${MASTER_ENDPOINTS_RESOLVER.key} to resolve the master endpoints. By default Celeborn uses " +
        "`org.apache.celeborn.common.client.StaticMasterEndpointResolver` which take static master endpoints " +
        "as input. Allowed pattern: `<host1>:<port1>[,<host2>:<port2>]*`, e.g. `clb1:9097,clb2:9098,clb3:9099`. " +
        "If the port is omitted, 9097 will be used. If the master endpoints are not static then users can pass " +
        s"custom resolver implementation to discover master endpoints actively using ${MASTER_ENDPOINTS_RESOLVER.key}.")
      .version("0.2.0")
      .stringConf
      .toSequence
      .createWithDefaultString(s"<localhost>:9097")
```

Reply:
@s0nskar, when provided resolver class isn't present in the classpath, could you add this check to help users discover this mismatching?

Reply:
I've added a new check to verify if resolver class is loadable. Which throws "Resolver class was not found in the classpath. Please check the class name and ensure that it is present in classpath" error if class is not found.

> give the default implementation for this situation?
IMO this can lead to unexpected behaviour during runtime, so its better we fast fail incase resolve class is not found.